packages/app/src/components/Table/Pagination.tsx

```diff
@@ -30,42 +35,67 @@ const Pagination: FC<PaginationProps> = React.memo(
 		setPage,
 		nextPage,
 		previousPage,
+		variant = 'default',
+		extra,
 	}) => {
 		const { t } = useTranslation()
 
 		const firstPage = () => setPage(0)
 		const toLastPage = () => setPage(pageCount - 1)
 
-		return (
-			<PaginationContainer compact={compact}>
-				<span>
-					<ArrowButton onClick={firstPage} disabled={!canPreviousPage}>
-						<LeftEndArrowIcon />
-					</ArrowButton>
-					<ArrowButton onClick={previousPage} disabled={!canPreviousPage}>
-						<LeftArrowIcon />
-					</ArrowButton>
-				</span>
-				<PageInfo>
-					{t('common.pagination.page')}{' '}
-					{t('common.pagination.page-of-total-pages', {
-						page: pageIndex + 1,
-						totalPages: pageCount,
-					})}
-				</PageInfo>
-				<span>
-					<ArrowButton onClick={nextPage} disabled={!canNextPage}>
-						<RightArrowIcon />
-					</ArrowButton>
-					<ArrowButton onClick={toLastPage} disabled={!canNextPage}>
-						<RightEndArrowIcon />
-					</ArrowButton>
-				</span>
-			</PaginationContainer>
+		return variant === 'default' ? (
+			<>
+				<PaginationContainer compact={compact}>
+					<ArrowButtonContainer>
+						<ArrowButton onClick={firstPage} disabled={!canPreviousPage}>
+							<LeftEndArrowIcon />
+						</ArrowButton>
+						<ArrowButton onClick={previousPage} disabled={!canPreviousPage}>
+							<LeftArrowIcon />
+						</ArrowButton>
+					</ArrowButtonContainer>
+					<PageInfo>
+						{t('common.pagination.page')}{' '}
+						{t('common.pagination.page-of-total-pages', {
+							page: pageIndex + 1,
+							totalPages: pageCount,
+						})}
+					</PageInfo>
+					<ArrowButtonContainer>
+						<ArrowButton onClick={nextPage} disabled={!canNextPage}>
+							<RightArrowIcon />
+						</ArrowButton>
+						<ArrowButton onClick={toLastPage} disabled={!canNextPage}>
+							<RightEndArrowIcon />
+						</ArrowButton>
+					</ArrowButtonContainer>
+				</PaginationContainer>
+				{extra}
+			</>
+		) : (
+			<StakingPagination
```

Comment:
This isn't really what I was hoping for but going to chat with Yashar about the differences here.